### PR TITLE
Deploy from Release Pipeline

### PR DESCRIPTION
**Pull request by balteravishay**
This PR adds the deployment stage to the release pipeline.
* extract the jobs to new template
* add required pipeline parameters
* documentation about the CI/CD process

link to release pipeline run:
https://dev.azure.com/csedevil/Presidio/_build/results?buildId=10742&view=results

link to CI pipeline run:
https://dev.azure.com/csedevil/Presidio/_build/results?buildId=10736&view=results

### CODE REVIEW

File: .pipelines/templates/release.yml [hunk 1]
```diff
@@ -5,64 +5,81 @@ pr: none
...
+            TAG: ":v2-latest"
+            AZURE_SUBSCRIPTION: $(ACR_AZURE_SUBSCRIPTION)
+  - stage: DeployProd
+    dependsOn: Release
+    variables:
+      REGISTRY_NAME: 'mcr.microsoft.com/'
+      VERSION: $[ stageDependencies.Release.version.outputs['setVer.version'] ]
+    displayName: Deploy to Production environment
+    jobs:
+      - template: ./deploy-stage.yml
```
Comment (SharonHart) [inline .pipelines/templates/release.yml:77]: I would just rename this to plain deploy

**Review by SharonHart — APPROVED** [review]

File: .pipelines/templates/release.yml (hunk 1 above)
Comment (balteravishay) [reply .pipelines/templates/release.yml:78]: done

**Review by SharonHart — DISMISSED** [review]

File: docs/build_release.md [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+Presidio leverages Azure DevOps YAML pipelines to validate, build, releasee and deliver presidio. the pipelines make use of [templates](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops) for code reuse using [YAML Schema](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema).
+
 ## Description
 
-## Table of contents
+The following pipelines are provided and maintained as part of presidio development process:
+
+* [PR Validation](../azure-pipelines.yml) - used to validate pull requests.
+    * Linting
+    * Credential Scanning
```
Comment (navalev) [inline docs/build_release.md:11]: we also do Component Governance detection and code analysis. This item should be just names security and compliance analysis

**Review by navalev — CHANGES_REQUESTED** [review]
looks great. minor comment on the readme

File: docs/build_release.md (hunk 2 above)
Comment (balteravishay) [reply docs/build_release.md:11]: done

**Review by navalev — APPROVED** [review]

**Review by SharonHart — APPROVED** [review]

**Pull request merged**